### PR TITLE
Private memberships can be used for authorization

### PR DESCRIPTION
**Pull request by samrocketman**
Private organization memberships can now be used for authorization.

fixes [JENKINS-20845][JENKINS-20845]

[JENKINS-20845]: https://issues.jenkins-ci.org/browse/JENKINS-20845

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/GithubAuthenticationToken.java [hunk 1]
```diff
@@ -302,7 +302,14 @@ public GithubOAuthUserDetails getUserDetails(String username) {
         if (user != null) {
             List<GrantedAuthority> groups = new ArrayList<GrantedAuthority>();
             try {
-                for (GHOrganization ghOrganization : user.getOrganizations()) {
+                GHPersonSet<GHOrganization> orgs;
+                if(this.userName.equals(username)) {
```
Comment (samrocketman) [inline src/main/java/org/jenkinsci/plugins/GithubAuthenticationToken.java]: If GitHub scopes don't include the ability to read teams, then all authorizations for a user are wiped.  Need to fall back to the default method of determining authorizations publicly.

**Comment by samrocketman** [comment]
Should wait for #47 to be merged.  This should make use of the `hasScope()` function in the GithubSecurityRealm.

**Comment by daniel-beck** [comment]
I don't think anyone is monitoring that account.

**Comment by samrocketman** [comment]
@daniel-beck ah, I thought you were responding because I at mentioned that account.  I guess not :).  Well then I'll just chill out and wait for code reviews to come in.

**Comment by samrocketman** [comment]
This is now ready for code review.

**Pull request merged**